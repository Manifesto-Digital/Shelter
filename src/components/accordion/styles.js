import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
