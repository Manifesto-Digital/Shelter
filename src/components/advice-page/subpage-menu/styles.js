import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
