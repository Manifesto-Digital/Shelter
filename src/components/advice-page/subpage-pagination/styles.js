import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
