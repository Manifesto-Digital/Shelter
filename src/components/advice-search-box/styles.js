import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
