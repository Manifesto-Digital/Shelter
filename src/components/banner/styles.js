import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
