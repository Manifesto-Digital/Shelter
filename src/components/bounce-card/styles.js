import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  background-color: ${({ theme }) => theme.palette.offWhite};
  padding: ${({ theme }) => theme.spacing.standard};
  font-size: ${({ theme }) => theme.fontsize.small};

  img {
    width: 45px;
    height: 76px;
    margin-right: ${({ theme }) => theme.spacing.small};
  }

  div {
    white-space: pre-line;
  }

  p {
    margin-bottom: ${({ theme }) => theme.spacing.small};
  }

  a {
    text-decoration: none;
  }
`;
