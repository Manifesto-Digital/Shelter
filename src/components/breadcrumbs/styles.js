import styled from 'styled-components';
import LinkHandler from '../link-handler';

export const BreadcrumbList = styled.ul`
  margin: 0;
  padding: 0.8em 0;
`;

export const BreadcrumbListItem = styled.li`
  display: inline-block;
  list-style: none;
  margin: 0 0.4em 0 0;
  padding: 0;
  font-size: ${({ theme }) => theme.fontsize.small};

  &:after {
    content: '>';
    margin-left: 0.4em;
  }
  &:last-child {
    &:after {
      content: '';
    }
  }
`;

export const BreadcrumbAnchor = styled(LinkHandler)`
  text-decoration: none;
`;
