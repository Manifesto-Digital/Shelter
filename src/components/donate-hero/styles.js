import styled from 'styled-components';
import { Tab, TabList } from 'react-tabs';
import { removeListStyles } from '../styled/utils';
import CTA from '../cta';
import { breakpoint } from '../theme/breakpoint';
import { Container } from '../styled/containers';
import LinkButton from '../link-button';

export const StyledDonateHero = styled.section`
  ${breakpoint.gt.desktop`
    height: 34rem;
  `}
  position: relative;
  overflow: hidden;
`;

export const StyledImage = styled.img`
  ${breakpoint.gt.desktop`
    position: absolute;
    height: 100%;
    // TODO: Make this work in IE11
    object-fit: cover;
  `}
  width: 100%;
`;

export const StyledSliderBox = styled.div`
  ${breakpoint.gt.desktop`
    width: 70%;
    margin-top: 3rem;
  `}
  background: ${props => props.theme.palette.grey15};
  position: relative;
`;

export const StyledTabList = styled(TabList)`
  ${removeListStyles};
  display: flex;
`;

export const StyledTab = styled(Tab).attrs({
  selectedClassName: 'is-selected',
})`
  flex: 1 1 0;
  padding: ${props => props.theme.spacing.standard};
  text-align: center;

  &:not(.is-selected) {
    background: ${props => props.theme.palette.offWhite};
    color: ${props => props.theme.palette.grey60};
  }

  &:hover {
    color: ${props => props.theme.palette.primary};
  }
`;

export const StyledTabContent = styled.section`
  padding: ${props => props.theme.spacing.large};
`;

export const StyledDonateButton = styled(CTA).attrs({
  bg: 'donate',
  fullWidth: true,
})`
  margin-bottom: ${props => props.theme.spacing.large};
`;

export const StyledDescription = styled.p`
  margin-bottom: ${props => props.theme.spacing.large};
`;

export const StyledCollapsableArea = styled.div`
  max-height: ${props => (props.collapsed ? '0' : '5rem')};
  transition: max-height 0.2s ease-in-out;
  overflow: hidden;
`;

export const StyledContainer = styled(Container)`
  ${breakpoint.lt.desktop`
    padding: 0;
    max-width: none;
  `}
`;

export const OwnAmountButton = styled(LinkButton)`
  display: block;
  margin-bottom: ${props => props.theme.spacing.standard};
`;
