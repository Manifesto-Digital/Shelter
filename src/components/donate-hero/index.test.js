import React from 'react';
import Slider from 'rc-slider/lib/Slider';
import DonateHero from '.';
import { mountWithTheme, snapshotComponent } from '../../../__tests__/helpers';
import { StyledDonateButton, StyledCollapsableArea } from './styles';
import DonationForm from '../donation-banner/donation-form';

const createImage = (type, amount) => ({
  src: `${type}-${amount}-foo`,
  srcSet: `${type}-${amount}-foo 100w, ${type}-${amount}-bar 200w`,
  title: `Image for ${type} ${amount}`,
});

const createAmount = (type, amount) => ({
  image: createImage(type, amount),
  description: `Donate £${amount} ${type}.`,
});

const createTab = (type, defaultAmount) => ({
  defaultAmount,
  amounts: {
    1: createAmount(type, 1),
    2: createAmount(type, 2),
    3: createAmount(type, 3),
    4: createAmount(type, 4),
  },
});

const expectAmount = (wrapper, { value, description, image }) => {
  expect(wrapper.find('img').props()).toMatchObject(image);
  expect(wrapper.find(Slider).prop('value')).toBe(value);
  expect(wrapper.find(StyledDonateButton).text()).toBe(`Donate £${value}`);
  // REVIEW: Maybe make this more specific?
  expect(wrapper.text()).toEqual(expect.stringContaining(description));
};

describe('error handling', () => {
  it('errors if default amount is not valid (once)', () => {
    expect(() => {
      mountWithTheme(
        <DonateHero
          monthly={createTab('monthly', 3)}
          once={createTab('once', 10)}
        />
      );
    }).toThrowErrorMatchingInlineSnapshot(
      `"Invalid default once amount of £10. The default amount must correspond to one of the specified amounts."`
    );
  });

  it('errors if default amount is not valid (monthly)', () => {
    expect(() => {
      mountWithTheme(
        <DonateHero
          monthly={createTab('monthly', 10)}
          once={createTab('once', 3)}
        />
      );
    }).toThrowErrorMatchingInlineSnapshot(
      `"Invalid default monthly amount of £10. The default amount must correspond to one of the specified amounts."`
    );
  });
});

it('renders correctly', () => {
  snapshotComponent(
    <DonateHero monthly={createTab('monthly', 2)} once={createTab('once', 3)} />
  );
});

it('initially displays default single-payment amount, background image and description', () => {
  const wrapper = mountWithTheme(
    <DonateHero monthly={createTab('monthly', 2)} once={createTab('once', 3)} />
  );

  const expectedImage = createImage('once', 3);

  expectAmount(wrapper, {
    value: 3,
    description: 'Donate £3 once.',
    image: expectedImage,
  });
});

it('displays default monthly amount, background image and description when switching to monthly tab', () => {
  const wrapper = mountWithTheme(
    <DonateHero monthly={createTab('monthly', 2)} once={createTab('once', 3)} />
  );

  wrapper.find('li[children="Give monthly"]').simulate('click');

  const expectedImage = createImage('monthly', 2);

  expectAmount(wrapper, {
    value: 2,
    description: 'Donate £2 monthly.',
    image: expectedImage,
  });
});

it('switches background image and description when changing slider', () => {
  const wrapper = mountWithTheme(
    <DonateHero monthly={createTab('monthly', 2)} once={createTab('once', 3)} />
  );

  const expectedImage = createImage('once', 1);

  const onChange = wrapper.find(Slider).prop('onChange');

  onChange(1);

  wrapper.update();

  expectAmount(wrapper, {
    value: 1,
    description: 'Donate £1 once.',
    image: expectedImage,
  });
});

it('renders a donation form', () => {
  const wrapper = mountWithTheme(
    <DonateHero monthly={createTab('monthly', 2)} once={createTab('once', 3)} />
  );

  expect(wrapper.find(DonationForm)).toHaveLength(1);
});

it('starts with donation form collapsed', () => {
  const wrapper = mountWithTheme(
    <DonateHero monthly={createTab('monthly', 2)} once={createTab('once', 3)} />
  );

  expect(wrapper.find(StyledCollapsableArea).prop('collapsed')).toBe(true);
});

it('expands donation form when clicking "Choose your own amount"', () => {
  const wrapper = mountWithTheme(
    <DonateHero monthly={createTab('monthly', 2)} once={createTab('once', 3)} />
  );

  wrapper
    .find({ children: 'Choose your own amount' })
    .first()
    .simulate('click');

  expect(wrapper.find(StyledCollapsableArea).prop('collapsed')).toBe(false);
});

describe('submission', () => {
  it('submits default amount to correct url', () => {
    const wrapper = mountWithTheme(
      <DonateHero
        monthly={createTab('monthly', 2)}
        once={createTab('once', 3)}
      />
    );

    expect(wrapper.find(StyledDonateButton).getDOMNode()).toHaveProperty(
      'href',
      'https://donate.shelter.org.uk/?cid=263&amount=300&frequency=once'
    );
  });
});
