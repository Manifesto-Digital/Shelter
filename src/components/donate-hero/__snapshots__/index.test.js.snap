// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c8 {
  position: relative;
  padding-top: 45px;
  margin-bottom: calc( 40px + 10px );
}

.c2 {
  max-width: 960px;
  margin: 0 auto;
  padding: 0 20px;
}

.c11 {
  display: inline-block;
}

.c10 {
  display: inline-block;
  text-align: center;
  padding: 10px 1em;
  font-weight: normal;
  background-color: #12b9bf;
  border: none;
  color: #fff;
  width: 100%;
}

.c0 {
  height: 34rem;
  position: relative;
  overflow: hidden;
}

.c1 {
  position: absolute;
  width: 100%;
}

.c3 {
  width: 70%;
  background: #d9d9d9;
  margin-top: 3rem;
  position: relative;
}

.c4 {
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 > li {
  margin: 0;
}

.c5 {
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  padding: 20px;
  text-align: center;
}

.c5:not(.is-selected) {
  background: #f7f7f7;
  color: #999;
}

.c5:hover {
  color: #ea232d;
}

.c6 {
  padding: 40px;
}

.c9 {
  margin-bottom: 40px;
}

.c7 {
  margin-bottom: 40px;
}

.c13 {
  max-height: 0;
  -webkit-transition: max-height 0.2s ease-in-out;
  transition: max-height 0.2s ease-in-out;
  overflow: hidden;
}

.c12 {
  padding: 0;
  border: none;
  font: inherit;
  color: inherit;
  background-color: transparent;
  cursor: pointer;
  color: #4d63b5;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 700;
}

.c12:hover {
  color: #ea232d;
}

.c16 {
  position: relative;
}

.c18 {
  display: inline-block;
  background-size: 15px 15px;
  width: auto;
  margin-bottom: 0;
  padding: 10px;
  padding-left: 30px;
  max-width: 10em;
}

.c18::-webkit-outer-spin-button,
.c18::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c17 {
  left: 12px;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  z-index: 1;
  font-size: 0.875rem;
}

.c15 {
  padding: 0;
  margin: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  -webkit-clip-path: inset(1px);
  clip-path: inset(1px);
  width: 1px;
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
}

.c20 {
  display: inline-block;
  text-align: center;
  padding: 10px 1em;
  font-weight: normal;
  background-color: #12b9bf;
  border: none;
  color: #fff;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c14 .c19 {
  margin-left: 10px;
}

@media (min-width:40.0625em) {
  .c14 {
    -webkit-flex: 0 1 40%;
    -ms-flex: 0 1 40%;
    flex: 0 1 40%;
    margin-bottom: 0;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

<section
  className="c0"
>
  <img
    className="c1"
    src="once-3-foo"
    srcSet="once-3-foo 100w, once-3-bar 200w"
    title="Image for once 3"
  />
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <div
        className="react-tabs"
        data-tabs={true}
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <ul
          className="c4"
          role="tablist"
        >
          <li
            aria-controls="react-tabs-1"
            aria-disabled="false"
            aria-selected="true"
            className="c5 is-selected"
            id="react-tabs-0"
            role="tab"
            tabIndex="0"
          >
            Give once
          </li>
          <li
            aria-controls="react-tabs-3"
            aria-disabled="false"
            aria-selected="false"
            className="c5"
            id="react-tabs-2"
            role="tab"
            tabIndex={null}
          >
            Give monthly
          </li>
        </ul>
        <div
          aria-labelledby="react-tabs-0"
          className="react-tabs__tab-panel react-tabs__tab-panel--selected"
          id="react-tabs-1"
          role="tabpanel"
        >
          <section
            className="c6"
          >
            <h2>
              Your donations make a difference
            </h2>
            <p
              className="c7"
            >
              Donate £3 once.
            </p>
            <div
              className="rc-slider rc-slider-with-marks c8"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="rc-slider-rail"
                style={
                  Object {
                    "backgroundColor": "#bdbec0",
                    "borderRadius": 5,
                    "height": 10,
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
              <div
                className="rc-slider-track"
                style={
                  Object {
                    "backgroundColor": "#5a5a5a",
                    "borderRadius": 5,
                    "height": 10,
                    "left": "0%",
                    "position": "absolute",
                    "width": "66.66666666666666%",
                  }
                }
              />
              <div
                className="rc-slider-step"
              >
                <span
                  className="rc-slider-dot rc-slider-dot-active"
                  style={
                    Object {
                      "left": "0%",
                    }
                  }
                />
                <span
                  className="rc-slider-dot rc-slider-dot-active"
                  style={
                    Object {
                      "left": "33.33333333333333%",
                    }
                  }
                />
                <span
                  className="rc-slider-dot rc-slider-dot-active"
                  style={
                    Object {
                      "left": "66.66666666666666%",
                    }
                  }
                />
                <span
                  className="rc-slider-dot"
                  style={
                    Object {
                      "left": "100%",
                    }
                  }
                />
              </div>
              <div
                aria-disabled={false}
                aria-valuemax={4}
                aria-valuemin={1}
                aria-valuenow={3}
                className="rc-slider-handle"
                onBlur={[Function]}
                onKeyDown={[Function]}
                role="slider"
                style={
                  Object {
                    "backgroundColor": "white",
                    "border": "1px solid #d9d9d9",
                    "borderRadius": "50%",
                    "boxShadow": "inset 0 0 1px #fff, inset 0 1px 7px #ebebeb, 0 3px 6px -3px #bbb",
                    "height": 30,
                    "left": "66.66666666666666%",
                    "marginLeft": -15,
                    "marginTop": -10,
                    "position": "absolute",
                    "width": 30,
                  }
                }
                tabIndex={0}
              />
              <div
                className="rc-slider-mark"
              >
                <span
                  className="rc-slider-mark-text rc-slider-mark-text-active"
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                      "display": "inline-block",
                      "left": "0%",
                      "lineHeight": "35px",
                      "marginLeft": "-15.000000000000002%",
                      "position": "absolute",
                      "textAlign": "center",
                      "top": 0,
                      "verticalAlign": "middle",
                      "width": "30.000000000000004%",
                    }
                  }
                >
                  £1
                </span>
                <span
                  className="rc-slider-mark-text rc-slider-mark-text-active"
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                      "display": "inline-block",
                      "left": "33.33333333333333%",
                      "lineHeight": "35px",
                      "marginLeft": "-15.000000000000002%",
                      "position": "absolute",
                      "textAlign": "center",
                      "top": 0,
                      "verticalAlign": "middle",
                      "width": "30.000000000000004%",
                    }
                  }
                >
                  £2
                </span>
                <span
                  className="rc-slider-mark-text rc-slider-mark-text-active"
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                      "display": "inline-block",
                      "fontSize": "1.25em",
                      "fontWeight": "bold",
                      "left": "66.66666666666666%",
                      "lineHeight": "35px",
                      "marginLeft": "-15.000000000000002%",
                      "position": "absolute",
                      "textAlign": "center",
                      "top": 0,
                      "verticalAlign": "middle",
                      "width": "30.000000000000004%",
                    }
                  }
                >
                  £3
                </span>
                <span
                  className="rc-slider-mark-text"
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                      "display": "inline-block",
                      "left": "100%",
                      "lineHeight": "35px",
                      "marginLeft": "-15.000000000000002%",
                      "position": "absolute",
                      "textAlign": "center",
                      "top": 0,
                      "verticalAlign": "middle",
                      "width": "30.000000000000004%",
                    }
                  }
                >
                  £4
                </span>
              </div>
            </div>
            <a
              className="c9 c10"
              href="https://donate.shelter.org.uk/?cid=263&amount=300&frequency=once"
            >
              <span
                className="c11"
              >
                Donate £
                3
              </span>
            </a>
            <button
              className="c12"
              onClick={[Function]}
            >
              Choose your own amount
            </button>
            <div
              className="c13"
            >
              <form
                action="https://donate.shelter.org.uk/b"
                className="c14"
                method="GET"
                onReset={[Function]}
                onSubmit={[Function]}
              >
                <input
                  name="cid"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="hidden"
                  value="263"
                />
                <input
                  name="free_amount"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="hidden"
                  value="1"
                />
                <input
                  name="amount"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="hidden"
                  value="3000"
                />
                <input
                  name="reserved_appeal_code"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="hidden"
                  value=""
                />
                <input
                  name="frequency"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="hidden"
                  value="once"
                />
                <label
                  className="c15"
                  htmlFor="amount-holder"
                >
                  Donate
                </label>
                <div
                  className="c16"
                >
                  <span
                    className="c17"
                  >
                    £
                  </span>
                  <input
                    className="c18"
                    id="amount-holder"
                    name="amount-holder"
                    onBlur={[Function]}
                    onChange={[Function]}
                    pattern="[0-9]*"
                    placeholder="30"
                    type="number"
                    value=""
                  />
                </div>
                <button
                  className="c19 c20"
                  type="submit"
                >
                  Donate
                </button>
              </form>
            </div>
          </section>
        </div>
        <div
          aria-labelledby="react-tabs-2"
          className="react-tabs__tab-panel"
          id="react-tabs-3"
          role="tabpanel"
        />
      </div>
    </div>
  </div>
</section>
`;
